R/impute-lag.R

```diff
@@ -0,0 +1,61 @@
+#' Impute the spatial lag of neighbors
+#'
+#' The `st_impute_lag` function computes the spatial lag for a given variable by
+#' replacing missing (`NA`) values using a user-specified function applied to
+#' neighbors' values (default function is mean). The imputation is done selectively for `NA` values only,
+#' retaining non-`NA` entries of the input vector `x`.
+#'
+#' @param x A numeric vector of spatial data with potential `NA` values to be imputed.
+#' @param nb A neighbor list object indicating the spatial relationships
+#'    between observations. Typically, this could be generated by
+#'    `sfdep::st_contiguity()`, `sfdep::st_knn()` or other spatial-dependency identifying methods.
+#' @param FUN A function to be applied to compute the spatial lag, e.g., `mean`,
+#'    `median`, etc. Default is `mean`.
+#' @param na_okay Logical, defaulting to `FALSE`. If `TRUE`, allows the presence
+#'    of `NA` values in `x` during the imputation. If `FALSE`, any `NA` value
+#'    encountered will generate an error.
+#' @param allow_zero A logical value or `NULL`. If `TRUE`, assigns zero as
+#'    lagged value to zones without neighbors. If `FALSE`, zones without
+#'    neighbors will generate an error. Default is `NULL`.
+#' @param ... Additional parameters passed to underlying functions.
+#'
+#' @details
+#' The spatial lag imputation is performed only on `NA` values of the input
+#' vector `x`. The imputation is based on the spatial relationships specified
+#' by the `nb` parameter, using the mean value of the neighboring zones for
+#' imputation. This ensures that the imputed value is spatially coherent with
+#' the nearby observations.
+#'
+#' @examples
+#' \dontrun{
```

Comment:
Can't use `\dontrun` in roxygen anymore. 
Since this uses mutate it should be wrapped in `if (rlang::is_installed("dplyr"))` to make sure the example can be ran with dplyr. 
Please use the build in guerry data set so you don't have to read nc. See other examples for reference.